tests/scripts/check_names.py

```diff
@@ -457,6 +457,105 @@ def parse_enum_consts(self, include, exclude=None):
 
         return enum_consts
 
+    IDENTIFIER_REGEX = re.compile(
+        # Match " something(a" or " *something(a". Functions.
+        # Assumptions:
+        # - function definition from return type to one of its arguments is
+        #   all on one line
+        # - function definition line only contains alphanumeric, asterisk,
+        #   underscore, and open bracket
+        r".* \**(\w+) *\( *\w|"
+        # Match "(*something)(".
+        r".*\( *\* *(\w+) *\) *\(|"
+        # Match names of named data structures.
+        r"(?:typedef +)?(?:struct|union|enum) +(\w+)(?: *{)?$|"
+        # Match names of typedef instances, after closing bracket.
+        r"}? *(\w+)[;[].*"
+    )
+    # The regex below is indented for clarity.
+    EXCLUSION_LINES = re.compile(
+        r"^("
+            r"extern +\"C\"|" # pylint: disable=bad-continuation
+            r"(typedef +)?(struct|union|enum)( *{)?$|"
+            r"} *;?$|"
+            r"$|"
+            r"//|"
+            r"#"
+        r")"
+    )
+
+    def parse_identifiers_in_file(self, header_file, identifiers):
+        """
+        Parse all lines of a header where a function/enum/struct/union/typedef
+        identifier is declared, based on some regex and heuristics. Highly
+        dependent on formatting style.
+
+        Append found matches to the list ``identifiers``.
+        """
+
+        with open(header_file, "r", encoding="utf-8") as header:
+            in_block_comment = False
+            # The previous line variable is used for concatenating lines
+            # when identifiers are formatted and spread across multiple
+            # lines.
+            previous_line = ""
+
+            for line_no, line in enumerate(header):
+                # Terminate current comment?
+                if in_block_comment:
+                    line = re.sub(r".*?\*/", r"", line, 1)
+                    in_block_comment = False
+                # Remove full comments and string literals
+                line = re.sub(r'/\*.*?\*/|(")(?:[^\\\"]|\\.)*"',
+                              lambda s: '""' if s.group(1) else ' ',
+                              line)
+                # Start an unfinished comment?
+                m = re.match(r"/\*", line)
+                if m:
+                    in_block_comment = True
+                    line = line[:m.end(0)]
```

Comment:
This retains the `/*` - e.g. `func(); /* call func` turns into `func(); /*`. Is that intended - I don't see why this would be retained?